src/pages/billing/Project.js

```diff
@@ -19,20 +20,76 @@ import * as Utils from 'src/libs/utils'
 import { billingRoles } from 'src/pages/billing/List'
 
 
-const workspaceLastModifiedWidth = 200
-const styles = {
-  workspaceCardField: {
-    flex: 1, width: `calc(50% - ${workspaceLastModifiedWidth / 2}px)`, whiteSpace: 'nowrap', textOverflow: 'ellipsis', overflow: 'hidden'
+const workspaceLastModifiedWidth = 150
+
+const WorkspaceCardHeaders = Utils.memoWithName('WorkspaceCardHeaders', ({ sort, onSort }) => {
+  return div({ role: 'row', style: { display: 'flex', justifyContent: 'space-between', marginTop: '1.5rem', padding: '0 1rem', marginBottom: '0.5rem' } }, [
+    div({ role: 'columnheader', 'aria-sort': ariaSort(sort, 'name'), style: { flex: 1 } }, [
+      h(HeaderRenderer, { sort, onSort, name: 'name' })
+    ]),
+    div({ role: 'columnheader', 'aria-sort': ariaSort(sort, 'createdBy'), style: { flex: 1 } }, [
+      h(HeaderRenderer, { sort, onSort, name: 'createdBy' })
+    ]),
+    div({ role: 'columnheader', 'aria-sort': ariaSort(sort, 'lastModified'), style: { flex: `0 0 ${workspaceLastModifiedWidth}px` } }, [
+      h(HeaderRenderer, { sort, onSort, name: 'lastModified' })
+    ])
+  ])
+})
+
+const rowBase = { display: 'flex', alignItems: 'center', width: '100%' }
+
+const ExpandedInfoRow = Utils.memoWithName('ExpandedInfoRow', ({ title, details, additionalInfo }) => {
+  const expandedInfoStyles = {
+    row: { ...rowBase, marginTop: '0.5rem' },
+    title: { fontWeight: 600, width: '20%', paddingRight: '1rem' },
+    details: { flexGrow: 1, width: '20%', paddingRight: '1rem', ...Style.noWrapEllipsis },
+    additionalInfo: { flexGrow: 1 }
   }
-}
 
-const WorkspaceCard = Utils.memoWithName('WorkspaceCard', ({ workspace }) => div({
-  style: Style.cardList.longCardShadowless
-}, [
-  div({ style: styles.workspaceCardField }, [workspace.name]),
-  div({ style: styles.workspaceCardField }, [workspace.createdBy]),
-  div({ style: { flex: `0 0 ${workspaceLastModifiedWidth}px` } }, [workspace.lastModified])
-]))
+  return div({ style: expandedInfoStyles.row, role: 'row' }, [
+    div({ role: 'cell', style: expandedInfoStyles.title }, [title]),
```

Comment:
@andrewmarcus, this is the one I most don't know about. Semantically, should this be a cell or a row header?

Reply:
Could go either way. I made these `rowheader`s in my version because it seemed that the title pretty cleanly described the row, and was generally unique within the set.

If you do include a rowheader, then users of assistive technologies will hear both the row header and column header on each cell, which helps with orientation. I didn't do this in the case of the general tables though, since I wasn't sure the first column would always be the row header. I probably should go back and add that, though, now that I think about it.

Reply:
It appears you're trying to create an expandable section within a row. You can't just nest `row`s inside each other, and you can't add more `cell`s to a row than there are `columnheader`s. These roles essentially just replicate `<table>`, `<tr>`, `<th>` and `<td>`.

Because of the expanding interaction, it probably no longer makes sense to use table semantics. You could try `treegrid`, or grouping these cards into `rowgroup`s with extra columns for the inner tables. But all of those are much more complicated than this really needs.

I conferred with a colleague. We recommend it's probably best to just go back to normal list semantics, where each list item has a button to expand or collapse the card.

```
<div role="list">
  <div role="listitem">
    <button type="button" aria-expanded="true" aria-controls="id1">2019_ASGH</button>
    <div>cloreth@broadinstitute.org</div>
    <div>Jan 25, 2021</div>
    <div id="id1">
        <div>Google project</div>
        ...
    </div>
  </div>
</div>
```

The sortable column headers will be interesting. But I think they are already buttons and already have ARIA descriptions telling users to click them to sort. That's likely good enough.

Reply:
I added similar semantics to the `Collapse` component in one of the PRs, I think it was #2526. You might be able to reuse that component here, or at least do the same thing it does